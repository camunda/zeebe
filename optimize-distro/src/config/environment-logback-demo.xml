<configuration>
  <conversionRule conversionWord="sanitize" converterClass="io.camunda.optimize.util.LogUtil" />
  <if condition='property("JSON_LOGGING").contains("true")'>
    <then>
      <appender name="UPGRADE_JSON" class="ch.qos.logback.core.ConsoleAppender">
        <encoder class="net.logstash.logback.encoder.LogstashEncoder">
        </encoder>
      </appender>
      <appender name="RUNTIME_JSON" class="ch.qos.logback.core.rolling.RollingFileAppender">
        <file>log/optimize.log</file>

        <rollingPolicy class="ch.qos.logback.core.rolling.SizeAndTimeBasedRollingPolicy">
          <fileNamePattern>log/optimize.%d{yyyy-MM-dd}.%i.log</fileNamePattern>
          <maxFileSize>50MB</maxFileSize>
          <maxHistory>30</maxHistory>
          <totalSizeCap>1GB</totalSizeCap>
        </rollingPolicy>

        <encoder class="net.logstash.logback.encoder.LogstashEncoder">
        </encoder>
      </appender>
      <!-- Change the log level for the upgrade here. -->
      <logger name="io.camunda.optimize" level="info"/>
      <logger name="org.elasticsearch" level="warn"/>
      <logger name="io.camunda.optimize.upgrade" level="info" additivity="false">
        <appender-ref ref="UPGRADE_JSON"/>
      </logger>
      <root level="warn">
        <appender-ref ref="RUNTIME_JSON"/>
      </root>
    </then>
    <else>
      <appender name="UPGRADE" class="ch.qos.logback.core.ConsoleAppender">
        <encoder>
          <pattern>%d{HH:mm:ss.SSS} - %sanitize(%msg) %n</pattern>
        </encoder>
      </appender>
      <appender name="RUNTIME" class="ch.qos.logback.core.rolling.RollingFileAppender">
        <file>log/optimize.log</file>

        <rollingPolicy class="ch.qos.logback.core.rolling.SizeAndTimeBasedRollingPolicy">
          <fileNamePattern>log/optimize.%d{yyyy-MM-dd}.%i.log</fileNamePattern>
          <maxFileSize>50MB</maxFileSize>
          <maxHistory>30</maxHistory>
          <totalSizeCap>1GB</totalSizeCap>
        </rollingPolicy>

        <encoder>
          <pattern>%d{HH:mm:ss.SSS} [%thread] %-5level %logger{36} - %sanitize(%msg) %n</pattern>
        </encoder>
      </appender>
      <!-- Change the log level for the upgrade here. -->
      <logger name="io.camunda.optimize" level="info"/>
      <logger name="org.elasticsearch" level="warn"/>
      <logger name="io.camunda.optimize.upgrade" level="info" additivity="false">
        <appender-ref ref="UPGRADE"/>
      </logger>
      <root level="warn">
        <appender-ref ref="RUNTIME"/>
      </root>
    </else>
  </if>
</configuration>