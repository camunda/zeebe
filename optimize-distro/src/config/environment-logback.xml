<configuration>
  <conversionRule conversionWord="sanitize" converterClass="io.camunda.optimize.util.LogUtil" />
  <if condition='property("JSON_LOGGING").contains("true")'>
    <then>
      <appender name="UPGRADE_JSON" class="ch.qos.logback.core.ConsoleAppender">
        <encoder class="net.logstash.logback.encoder.LogstashEncoder">
        </encoder>
      </appender>
      <appender name="RUNTIME_JSON" class="ch.qos.logback.core.ConsoleAppender">
        <encoder class="net.logstash.logback.encoder.LogstashEncoder">
        </encoder>
      </appender>
      <!-- Change the log level for the upgrade here. If provided, the environment variable will take precedence -->
      <logger name="io.camunda.optimize" level="${OPTIMIZE_LOG_LEVEL:-info}"/>
      <logger name="org.elasticsearch" level="${ES_LOG_LEVEL:-warn}"/>
      <logger name="io.camunda.optimize.upgrade" level="${UPGRADE_LOG_LEVEL:-info}" additivity="false">
        <appender-ref ref="UPGRADE_JSON"/>
      </logger>
      <root level="warn">
        <appender-ref ref="RUNTIME_JSON"/>
      </root>
    </then>
    <else>
      <appender name="UPGRADE" class="ch.qos.logback.core.ConsoleAppender">
        <encoder>
          <pattern>%d{HH:mm:ss.SSS} - %sanitize(%msg) %n</pattern>
        </encoder>
      </appender>
      <appender name="RUNTIME" class="ch.qos.logback.core.ConsoleAppender">
        <encoder>
          <pattern>%d{HH:mm:ss.SSS} [%thread] %-5level %logger{36} - %sanitize(%msg) %n</pattern>
        </encoder>
      </appender>
      <!-- Change the log level for the upgrade here. If provided, the environment variable will take precedence -->
      <logger name="io.camunda.optimize" level="${OPTIMIZE_LOG_LEVEL:-info}"/>
      <logger name="org.elasticsearch" level="${ES_LOG_LEVEL:-warn}"/>
      <logger name="io.camunda.optimize.upgrade" level="${UPGRADE_LOG_LEVEL:-info}" additivity="false">
        <appender-ref ref="UPGRADE"/>
      </logger>
      <root level="warn">
        <appender-ref ref="RUNTIME"/>
      </root>
    </else>
  </if>
</configuration>