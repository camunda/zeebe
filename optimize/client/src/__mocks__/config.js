/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a proprietary license.
 * See the License.txt file for more information. You may not use this file
 * except in compliance with the proprietary license.
 */

export const isEmailEnabled = () => true;
export const isSharingEnabled = () => true;
export const getOptimizeVersion = () => '2.7.0';
export const getWebappEndpoints = () => ({
  default: {endpoint: 'http://localhost:8080/camunda', engineName: 'default'},
});
export const getHeader = () => ({
  textColor: 'dark', // or "light"
  backgroundColor: '#FFFFFF',
  logo: '',
});
export const ConfigProvider = () => <div />;
export {default as newReport} from '../modules/newReport.json';
