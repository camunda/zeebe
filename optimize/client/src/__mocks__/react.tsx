/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a proprietary license.
 * See the License.txt file for more information. You may not use this file
 * except in compliance with the proprietary license.
 */

// @ts-ignore
export * from 'react';
export {default} from 'react';

const outstandingEffects: (() => void)[] = [];
const outstandingCleanups: (() => void)[] = [];

export const useEffect = (fn: () => (() => void) | void) => outstandingEffects.push(fn);
export const runLastEffect = () => {
  if (outstandingEffects.length) {
    const cleanup = outstandingEffects.pop()?.();

    if (cleanup) {
      outstandingCleanups.push(cleanup);
    }
  }
};

export const runLastCleanup = () => {
  if (outstandingCleanups.length) {
    outstandingCleanups.pop()?.();
  }
};

export const runAllEffects = () => {
  const numberOfEffects = outstandingEffects.length;

  for (let i = 0; i < numberOfEffects; i++) {
    const cleanup = outstandingEffects.shift()?.();

    if (cleanup) {
      outstandingCleanups.push(cleanup);
    }
  }
};

beforeEach(() => {
  outstandingEffects.length = 0;
});

export const runAllCleanups = () => {
  const numberOfEffects = outstandingCleanups.length;

  for (let i = 0; i < numberOfEffects; i++) {
    outstandingCleanups.shift()?.();
  }
};
