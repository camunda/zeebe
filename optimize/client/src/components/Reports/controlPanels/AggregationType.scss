.AggregationType {
  .ButtonTrigger {
    color: var(--cds-text-primary, #161616);
  }
}
