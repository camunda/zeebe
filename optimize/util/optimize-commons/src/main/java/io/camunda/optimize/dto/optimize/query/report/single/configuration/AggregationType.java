/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
 * one or more contributor license agreements. See the NOTICE file distributed
 * with this work for additional information regarding copyright ownership.
 * Licensed under the Camunda License 1.0. You may not use this file
 * except in compliance with the Camunda License 1.0.
 */
package io.camunda.optimize.dto.optimize.query.report.single.configuration;

import static io.camunda.optimize.dto.optimize.ReportConstants.AVERAGE_AGGREGATION_TYPE;
import static io.camunda.optimize.dto.optimize.ReportConstants.MAX_AGGREGATION_TYPE;
import static io.camunda.optimize.dto.optimize.ReportConstants.MIN_AGGREGATION_TYPE;
import static io.camunda.optimize.dto.optimize.ReportConstants.PERCENTILE_AGGREGATION_TYPE;
import static io.camunda.optimize.dto.optimize.ReportConstants.SUM_AGGREGATION_TYPE;

import com.fasterxml.jackson.annotation.JsonValue;

public enum AggregationType {
  AVERAGE(AVERAGE_AGGREGATION_TYPE),
  MIN(MIN_AGGREGATION_TYPE),
  MAX(MAX_AGGREGATION_TYPE),
  SUM(SUM_AGGREGATION_TYPE),
  PERCENTILE(PERCENTILE_AGGREGATION_TYPE);

  private final String id;

  AggregationType(final String id) {
    this.id = id;
  }

  @JsonValue
  public String getId() {
    return id;
  }

  @Override
  public String toString() {
    return getId();
  }
}
