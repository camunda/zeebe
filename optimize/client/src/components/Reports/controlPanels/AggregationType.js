/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a proprietary license.
 * See the License.txt file for more information. You may not use this file
 * except in compliance with the proprietary license.
 */

import {Edit} from '@carbon/icons-react';
import {Form, FormGroup, Stack, Toggle} from '@carbon/react';

import {Popover} from 'components';
import {t} from 'translation';

import './AggregationType.scss';

const orders = {
  aggregationTypes: ['sum', 'min', 'avg', 'max', 'percentile'],
  userTaskDurationTimes: ['total', 'work', 'idle'],
  percentileAggregations: [99, 95, 90, 75, 50, 25],
};

export default function AggregationType({report, onChange}) {
  const {configuration, distributedBy} = report;
  const {aggregationTypes, processPart} = report.configuration;

  const isDurationReport = report?.view?.properties.includes('duration');
  const isUserTaskReport = report?.view?.entity === 'userTask';
  const isVariableReport = report?.view?.entity === 'variable';
  const isIncidentReport = report?.view?.entity === 'incident';

  function isLastAggregation(field, type, value) {
    return (
      configuration[field].length === 1 &&
      getType(field, configuration[field][0]) === type &&
      sameValue(configuration[field][0], value)
    );
  }

  function hasAggregation(field, type, value) {
    return configuration[field].some(
      (agg) => getType(field, agg) === type && sameValue(agg, value)
    );
  }

  function addAggregation(field, type, value) {
    const newAggregations = [
      ...configuration[field],
      constructNewAggregation(field, type, value),
    ].sort((a, b) => {
      if (a.type === 'percentile' && b.type === 'percentile') {
        return b.value - a.value;
      }

      return orders[field].indexOf(getType(field, a)) - orders[field].indexOf(getType(field, b));
    });

    return updateReport(field, newAggregations, true);
  }

  function removeAggregation(field, type, value) {
    const remainingAggregations = configuration[field].filter(
      (existingAgg) =>
        getType(field, existingAgg) !== type ||
        (getType(field, existingAgg) === type && !sameValue(existingAgg, value))
    );

    return updateReport(field, remainingAggregations);
  }

  function updateReport(field, aggregations, resetTargetValue) {
    const changes = {
      configuration: {
        [field]: {
          $set: aggregations,
        },
      },
    };

    if (resetTargetValue) {
      changes.configuration.targetValue = {active: {$set: false}};
    }

    return onChange(changes, true);
  }

  if (isDurationReport || isVariableReport) {
    const availableAggregations = [];
    if (!isIncidentReport) {
      availableAggregations.push('sum');
    }
    availableAggregations.push('min', 'avg');
    availableAggregations.push('max');

    const {type, value} = aggregationTypes[0];
    let popoverTitle = t('report.config.aggregationShort.' + type, {value});

    if (
      availableAggregations.every((aggregation) =>
        hasAggregation('aggregationTypes', aggregation.type)
      )
    ) {
      popoverTitle = 'All';
    } else if (aggregationTypes.length > 1) {
      popoverTitle = 'Multi';
    }

    return (
      <Popover
        className="AggregationType"
        isTabTip
        trigger={
          <Popover.Button
            size="sm"
            kind="ghost"
            renderIcon={Edit}
            iconDescription={t('report.config.aggregation.label')}
          >
            {popoverTitle}
          </Popover.Button>
        }
        floating
      >
        <Form>
          <Stack gap={3}>
            {isUserTaskReport && (
              <FormGroup legendText={t('report.config.aggregation.userTaskLegend')}>
                <Stack gap={3}>
                  {orders.userTaskDurationTimes.map((type) => (
                    <Toggle
                      key={type}
                      id={type}
                      size="sm"
                      labelText={t('report.config.userTaskDuration.' + type)}
                      hideLabel
                      toggled={hasAggregation('userTaskDurationTimes', type)}
                      disabled={isLastAggregation('userTaskDurationTimes', type)}
                      onToggle={(checked) => {
                        if (checked) {
                          addAggregation('userTaskDurationTimes', type);
                        } else {
                          removeAggregation('userTaskDurationTimes', type);
                        }
                      }}
                    />
                  ))}
                </Stack>
              </FormGroup>
            )}
            <FormGroup
              legendText={t(
                'report.config.aggregation.' +
                  (isVariableReport ? 'variableLegend' : 'durationLegend')
              )}
            >
              <Stack gap={3}>
                {availableAggregations.map((type) => (
                  <Toggle
                    key={type}
                    id={type}
                    size="sm"
                    labelText={t('report.config.aggregation.' + type)}
                    hideLabel
                    toggled={hasAggregation('aggregationTypes', type)}
                    disabled={isLastAggregation('aggregationTypes', type)}
                    onToggle={(checked) => {
                      if (checked) {
                        addAggregation('aggregationTypes', type);
                      } else {
                        removeAggregation('aggregationTypes', type);
                      }
                    }}
                  />
                ))}
              </Stack>
            </FormGroup>
            {distributedBy.type !== 'process' && !processPart && (
              <FormGroup legendText={t('report.config.aggregation.percentileLegend')}>
                <Stack gap={3}>
                  {orders.percentileAggregations.map((value) => {
                    const defaultLabel = `P${value}`;
                    return (
                      <Toggle
                        key={defaultLabel}
                        id={defaultLabel}
                        size="sm"
                        labelText={value === 50 ? t('report.config.aggregation.p50') : defaultLabel}
                        hideLabel
                        toggled={hasAggregation('aggregationTypes', 'percentile', value)}
                        disabled={isLastAggregation('aggregationTypes', 'percentile', value)}
                        onToggle={(checked) => {
                          if (checked) {
                            addAggregation('aggregationTypes', 'percentile', value);
                          } else {
                            removeAggregation('aggregationTypes', 'percentile', value);
                          }
                        }}
                      />
                    );
                  })}
                </Stack>
              </FormGroup>
            )}
          </Stack>
        </Form>
      </Popover>
    );
  }
  return null;
}

function getType(field, aggregation) {
  if (field === 'userTaskDurationTimes') {
    return aggregation;
  }

  return aggregation.type;
}

function constructNewAggregation(field, type, value = null) {
  if (field === 'userTaskDurationTimes') {
    return type;
  }

  return {type, value};
}

function sameValue(agg, value) {
  if (!value) {
    return true;
  }

  return agg.value === value;
}
