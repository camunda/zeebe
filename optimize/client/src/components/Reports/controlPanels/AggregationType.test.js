/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a proprietary license.
 * See the License.txt file for more information. You may not use this file
 * except in compliance with the proprietary license.
 */

import {runAllEffects} from 'react';
import {shallow} from 'enzyme';

import {getOptimizeProfile} from 'config';

import AggregationType from './AggregationType';

jest.mock('config', () => ({
  getOptimizeProfile: jest.fn().mockReturnValue('platform'),
}));

it('should render nothing if the current result is no duration and the view is not variable', () => {
  const node = shallow(
    <AggregationType
      report={{
        configuration: {aggregationTypes: [{type: 'avg', value: null}]},
        view: {entity: null, properties: ['rawData']},
        distributedBy: {type: 'none'},
      }}
    />
  );

  expect(node).toMatchSnapshot();
});

it('should render an aggregation selection for duration reports', () => {
  const node = shallow(
    <AggregationType
      report={{
        view: {properties: ['duration']},
        distributedBy: {type: 'none'},
        configuration: {aggregationTypes: [{type: 'percentile', value: 50}]},
      }}
    />
  );

  expect(node).toMatchSnapshot();
});

it('should render a user task duration selection for user task duration reports', async () => {
  const node = shallow(
    <AggregationType
      report={{
        view: {entity: 'userTask', properties: ['duration']},
        distributedBy: {type: 'none'},
        configuration: {
          aggregationTypes: [{type: 'percentile', value: 50}],
          userTaskDurationTimes: ['idle'],
        },
      }}
    />
  );

  await runAllEffects();

  expect(node).toMatchSnapshot();
});

it('should render sum field for variable reports', () => {
  const node = shallow(
    <AggregationType
      report={{
        view: {entity: 'variable', properties: [{}]},
        distributedBy: {type: 'none'},
        configuration: {aggregationTypes: [{type: 'sum', value: null}]},
      }}
    />
  );

  expect(node.find('Toggle').first()).toHaveProp('labelText', 'Sum');
});

it('should hide sum field for incident reports', () => {
  const node = shallow(
    <AggregationType
      report={{
        view: {entity: 'incident', properties: ['duration']},
        distributedBy: {type: 'none'},
        configuration: {aggregationTypes: [{type: 'avg', value: null}]},
      }}
    />
  );

  expect(node.find('Toggle').first()).not.toHaveProp('labelText', 'Sum');
});

it('should reevaluate the report when changing the aggregation type', () => {
  const spy = jest.fn();

  const node = shallow(
    <AggregationType
      report={{
        view: {properties: ['duration']},
        distributedBy: {type: 'none'},
        configuration: {aggregationTypes: [{type: 'percentile', value: 50}]},
      }}
      onChange={spy}
    />
  );

  node.find('Toggle').last().simulate('toggle', true);

  expect(spy).toHaveBeenCalledWith(
    {
      configuration: {
        aggregationTypes: {
          $set: [
            {type: 'percentile', value: 50},
            {type: 'percentile', value: 25},
          ],
        },
        targetValue: {active: {$set: false}},
      },
    },
    true
  );
});

it('should hide percentile aggregations if processpart is defined', () => {
  const spy = jest.fn();

  const node = shallow(
    <AggregationType
      report={{
        view: {properties: ['duration']},
        distributedBy: {type: 'none'},
        configuration: {
          aggregationTypes: [{type: 'avg', value: null}],
          processPart: 'defined',
        },
      }}
      onChange={spy}
    />
  );

  expect(node.find({label: 'P25'})).not.toExist();
  expect(node.find({label: 'P50'})).not.toExist();
  expect(node.find({label: 'P95'})).not.toExist();
});

it('should not show user task duration selection for user task duration reports in cloud environment', async () => {
  getOptimizeProfile.mockReturnValueOnce('cloud');
  const node = shallow(
    <AggregationType
      report={{
        view: {entity: 'userTask', properties: ['duration']},
        distributedBy: {type: 'none'},
        configuration: {
          aggregationTypes: [{type: 'avg', value: null}],
          userTaskDurationTimes: ['total'],
        },
      }}
    />
  );

  await runAllEffects();

  expect(node.find({label: 'Work'})).not.toExist();
  expect(node.find({label: 'Idle'})).not.toExist();
});

it('should not show percentile aggregations when report is grouped by process ', async () => {
  const node = shallow(
    <AggregationType
      report={{
        view: {entity: 'processInstance', properties: ['duration']},
        distributedBy: {type: 'process'},
        configuration: {
          aggregationTypes: [{type: 'avg', value: null}],
        },
      }}
    />
  );

  await runAllEffects();

  expect(node.find({label: 'P25'})).not.toExist();
  expect(node.find({label: 'P95'})).not.toExist();
});
