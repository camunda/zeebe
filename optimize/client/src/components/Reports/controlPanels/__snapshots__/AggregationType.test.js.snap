// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a user task duration selection for user task duration reports 1`] = `
<Popover
  className="AggregationType"
  floating={true}
  isTabTip={true}
  trigger={
    <ButtonTrigger
      iconDescription="Aggregation"
      kind="ghost"
      renderIcon={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "propTypes": Object {
            "size": [Function],
          },
          "render": [Function],
        }
      }
      size="sm"
    >
      P50
    </ButtonTrigger>
  }
>
  <Form>
    <ForwardRef(Stack)
      gap={3}
    >
      <FormGroup
        legendText="User task duration"
      >
        <ForwardRef(Stack)
          gap={3}
        >
          <Toggle
            disabled={false}
            hideLabel={true}
            id="total"
            key="total"
            labelText="Total"
            onToggle={[Function]}
            size="sm"
            toggled={false}
          />
          <Toggle
            disabled={false}
            hideLabel={true}
            id="work"
            key="work"
            labelText="Assigned"
            onToggle={[Function]}
            size="sm"
            toggled={false}
          />
          <Toggle
            disabled={true}
            hideLabel={true}
            id="idle"
            key="idle"
            labelText="Unassigned"
            onToggle={[Function]}
            size="sm"
            toggled={true}
          />
        </ForwardRef(Stack)>
      </FormGroup>
      <FormGroup
        legendText="Basic aggregations"
      >
        <ForwardRef(Stack)
          gap={3}
        >
          <Toggle
            disabled={false}
            hideLabel={true}
            id="sum"
            key="sum"
            labelText="Sum"
            onToggle={[Function]}
            size="sm"
            toggled={false}
          />
          <Toggle
            disabled={false}
            hideLabel={true}
            id="min"
            key="min"
            labelText="Minimum"
            onToggle={[Function]}
            size="sm"
            toggled={false}
          />
          <Toggle
            disabled={false}
            hideLabel={true}
            id="avg"
            key="avg"
            labelText="Average"
            onToggle={[Function]}
            size="sm"
            toggled={false}
          />
          <Toggle
            disabled={false}
            hideLabel={true}
            id="max"
            key="max"
            labelText="Maximum"
            onToggle={[Function]}
            size="sm"
            toggled={false}
          />
        </ForwardRef(Stack)>
      </FormGroup>
      <FormGroup
        legendText="Percentile aggregations"
      >
        <ForwardRef(Stack)
          gap={3}
        >
          <Toggle
            disabled={false}
            hideLabel={true}
            id="P99"
            key="P99"
            labelText="P99"
            onToggle={[Function]}
            size="sm"
            toggled={false}
          />
          <Toggle
            disabled={false}
            hideLabel={true}
            id="P95"
            key="P95"
            labelText="P95"
            onToggle={[Function]}
            size="sm"
            toggled={false}
          />
          <Toggle
            disabled={false}
            hideLabel={true}
            id="P90"
            key="P90"
            labelText="P90"
            onToggle={[Function]}
            size="sm"
            toggled={false}
          />
          <Toggle
            disabled={false}
            hideLabel={true}
            id="P75"
            key="P75"
            labelText="P75"
            onToggle={[Function]}
            size="sm"
            toggled={false}
          />
          <Toggle
            disabled={true}
            hideLabel={true}
            id="P50"
            key="P50"
            labelText="P50 (median)"
            onToggle={[Function]}
            size="sm"
            toggled={true}
          />
          <Toggle
            disabled={false}
            hideLabel={true}
            id="P25"
            key="P25"
            labelText="P25"
            onToggle={[Function]}
            size="sm"
            toggled={false}
          />
        </ForwardRef(Stack)>
      </FormGroup>
    </ForwardRef(Stack)>
  </Form>
</Popover>
`;

exports[`should render an aggregation selection for duration reports 1`] = `
<Popover
  className="AggregationType"
  floating={true}
  isTabTip={true}
  trigger={
    <ButtonTrigger
      iconDescription="Aggregation"
      kind="ghost"
      renderIcon={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "propTypes": Object {
            "size": [Function],
          },
          "render": [Function],
        }
      }
      size="sm"
    >
      P50
    </ButtonTrigger>
  }
>
  <Form>
    <ForwardRef(Stack)
      gap={3}
    >
      <FormGroup
        legendText="Basic aggregations"
      >
        <ForwardRef(Stack)
          gap={3}
        >
          <Toggle
            disabled={false}
            hideLabel={true}
            id="sum"
            key="sum"
            labelText="Sum"
            onToggle={[Function]}
            size="sm"
            toggled={false}
          />
          <Toggle
            disabled={false}
            hideLabel={true}
            id="min"
            key="min"
            labelText="Minimum"
            onToggle={[Function]}
            size="sm"
            toggled={false}
          />
          <Toggle
            disabled={false}
            hideLabel={true}
            id="avg"
            key="avg"
            labelText="Average"
            onToggle={[Function]}
            size="sm"
            toggled={false}
          />
          <Toggle
            disabled={false}
            hideLabel={true}
            id="max"
            key="max"
            labelText="Maximum"
            onToggle={[Function]}
            size="sm"
            toggled={false}
          />
        </ForwardRef(Stack)>
      </FormGroup>
      <FormGroup
        legendText="Percentile aggregations"
      >
        <ForwardRef(Stack)
          gap={3}
        >
          <Toggle
            disabled={false}
            hideLabel={true}
            id="P99"
            key="P99"
            labelText="P99"
            onToggle={[Function]}
            size="sm"
            toggled={false}
          />
          <Toggle
            disabled={false}
            hideLabel={true}
            id="P95"
            key="P95"
            labelText="P95"
            onToggle={[Function]}
            size="sm"
            toggled={false}
          />
          <Toggle
            disabled={false}
            hideLabel={true}
            id="P90"
            key="P90"
            labelText="P90"
            onToggle={[Function]}
            size="sm"
            toggled={false}
          />
          <Toggle
            disabled={false}
            hideLabel={true}
            id="P75"
            key="P75"
            labelText="P75"
            onToggle={[Function]}
            size="sm"
            toggled={false}
          />
          <Toggle
            disabled={true}
            hideLabel={true}
            id="P50"
            key="P50"
            labelText="P50 (median)"
            onToggle={[Function]}
            size="sm"
            toggled={true}
          />
          <Toggle
            disabled={false}
            hideLabel={true}
            id="P25"
            key="P25"
            labelText="P25"
            onToggle={[Function]}
            size="sm"
            toggled={false}
          />
        </ForwardRef(Stack)>
      </FormGroup>
    </ForwardRef(Stack)>
  </Form>
</Popover>
`;

exports[`should render nothing if the current result is no duration and the view is not variable 1`] = `""`;
